Extension/src/LanguageServer/clientCollection.ts

```diff
@@ -215,3 +220,69 @@ export class ClientCollection {
         return cpptools.DefaultClient.stopLanguageClient();
     }
 }
+
+interface TimeStampSequence {
+    activationTime: number; // when the file appears in the editor. Defined for both "cold/warm" start cases.
+    setupTime: number; // when the Intellisense_client constructor is completed
+    updateRangeTime: number; // when publishDiagnostics & provideSemanticTokens is completed
+    totalTime: number;
+}
+
+export class TimeTelemetryCollector {
```

Comment:
Please move the collector class and dependent interfaces to a new file.  This doesn't belong in clientCollection.ts.